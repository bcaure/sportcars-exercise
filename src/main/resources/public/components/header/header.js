import getTemplate from '/components/util.js';

export default class HeaderComponent extends HTMLElement {

  get onMenuSelect() {
    return this._onMenuSelect;
  }

  set onMenuSelect(value) {
    this._onMenuSelect = value;
  }

  constructor() {
    super();
    this.attachShadow({mode: 'open'});
    getTemplate('/components/header/header.html').then(template => {
      this.shadowRoot.appendChild(template.content.cloneNode(true));
    this.shadowRoot.querySelector('.top10-weight').addEventListener('click',() => this._onMenuSelect('weight'));
    this.shadowRoot.querySelector('.top10-power').addEventListener('click', () => this._onMenuSelect('power'));
    this.shadowRoot.querySelector('.top10-ratio').addEventListener('click', () => this._onMenuSelect('ratio'));
  })
    ;

    this._onMenuSelect = function () {}; // callback defined by parent
  }

}

customElements.define('header-component', HeaderComponent);


