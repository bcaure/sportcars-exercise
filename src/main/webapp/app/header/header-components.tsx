import React from 'react';
import { DropdownMenu, DropdownToggle, NavbarBrand, NavItem, NavLink, UncontrolledDropdown } from 'reactstrap';
import { NavLink as Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

export const BrandIcon = props => (
  <div {...props} className="brand-icon">
    <img src="content/images/logo.jpg" alt="Logo" />
  </div>
);

export const Brand = props => (
  <NavbarBrand tag={Link} to="/" className="brand-logo">
    <BrandIcon />
    <span className="brand-title">Weight Cars</span>
  </NavbarBrand>
);

export const Home = props => (
  <NavItem>
    <NavLink tag={Link} to="/" className="d-flex align-items-center">
      <FontAwesomeIcon icon="home" />
      <span>Home</span>
    </NavLink>
  </NavItem>
);

export const Top10Weight = props => (
  <NavItem>
    <NavLink className="d-flex align-items-center" onClick={() => props.onClick()}>
      <FontAwesomeIcon icon="weight-hanging" />
      <span>TOP 10 Weight</span>
    </NavLink>
  </NavItem>
);

export const Top10Power = props => (
  <NavItem>
    <NavLink className="d-flex align-items-center" onClick={() => props.onClick()}>
      <FontAwesomeIcon icon="car-battery" />
      <span>TOP 10 Power</span>
    </NavLink>
  </NavItem>
);

export const Top10Ratio = props => (
  <NavItem>
    <NavLink className="d-flex align-items-center" onClick={() => props.onClick()}>
      <FontAwesomeIcon icon="stopwatch" />
      <span>TOP 10 Ratio</span>
    </NavLink>
  </NavItem>
);
