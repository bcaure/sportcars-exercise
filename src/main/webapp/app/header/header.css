@import '/static/bootstrap.min.css';
header {
  display: flex;
  align-items: center;
  height: 150px;
  padding: 50px;
}
.brand-icon img {
  width: 100px;
}
h1 {
  flex: 1;
  text-align: center;
}
