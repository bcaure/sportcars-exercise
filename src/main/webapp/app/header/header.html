<template>
    <link rel="stylesheet" href="app/header/header.css">
    <header id="app-header">

        <section class="brand-icon">
            <img src="static/images/logo.jpg" alt="Logo" />
        </section>
        <h1>Weight Cars</h1>

    </header>
</template>
<script src="header.js"></script>