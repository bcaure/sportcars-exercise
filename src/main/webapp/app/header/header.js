import getTemplate from '/app/util.js';

export class HeaderState {
  menuOpen;
}

export class HeaderProps {
  constructor(top10Click, menuOpen) {
    this.top10Click = top10Click;
    this.menuOpen = menuOpen;
  }
}

export default class HeaderComponent extends HTMLElement {
  state = {
    menuOpen: false
  };

  toggleMenu = () => {
    this.setState({ menuOpen: !this.state.menuOpen });
  };

  connectedCallback() {
    var shadow = this.attachShadow({mode: 'open'});
    this.render(shadow);
  }

  async render(shadow) {

    // Select the template and clone it. Finally attach the cloned node to the shadowDOM's root.
    // Current document needs to be defined to get DOM access to imported HTML
    const template = await getTemplate('/app/header/header.html');
    const instance = template.content.cloneNode(true);
    shadow.appendChild(instance);

  }
}

customElements.define('header-component', HeaderComponent);


