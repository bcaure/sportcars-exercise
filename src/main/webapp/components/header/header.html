<template>
  <link rel="stylesheet" href="components/header/header.css">
  <header class="app-header">

    <h1>Weight Cars</h1>
    <nav class="search">
      <input type="text" class="form-control" id="searchFilter" placeholder="Search" onKeyUp="filterCars()"/>
    </nav>


    <section class="nav-items">
      <details onclick="return false">
        <summary>
          <nav class="home">
            <span>Home</span>
          </nav>
        </summary>
      </details>
      <details class="menu-top10">
        <summary>
          <nav>
            <span>Top 10</span>
          </nav>
        </summary>

        <section>
          <nav class="top10-weight">
            <span>Weight</span>
          </nav>
          <nav class="top10-power">
            <span>Power</span>
          </nav>
          <nav class="top10-ratio">
            <span>Ratio</span>
          </nav>
        </section>
      </details>
    </section>

  </header>
</template>
<script src="header.js"></script>