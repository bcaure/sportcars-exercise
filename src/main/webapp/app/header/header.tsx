import './header.css';
import React from 'react';
import { Navbar, Nav, NavbarToggler, NavbarBrand, Collapse } from 'reactstrap';
import { Home, Brand } from './header-components';

export interface IHeaderState {
  menuOpen: boolean;
}

export default class Header extends React.Component<IHeaderState> {
  state: IHeaderState = {
    menuOpen: false
  };

  private toggleMenu = () => {
    this.setState({ menuOpen: !this.state.menuOpen });
  };

  public render() {
    return (
      <div id="app-header">
        <Navbar dark expand="sm" fixed="top" className="jh-navbar">
          <NavbarToggler aria-label="Menu" onClick={this.toggleMenu} />
          <Brand />
          <Collapse isOpen={this.state.menuOpen} navbar>
            <Nav id="header-tabs" className="ml-auto" navbar>
              <Home />
            </Nav>
          </Collapse>
        </Navbar>
      </div>
    );
  }
}
