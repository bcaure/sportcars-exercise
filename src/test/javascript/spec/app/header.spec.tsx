import React from 'react';
import { shallow } from 'enzyme';

import { Nav, Navbar } from 'reactstrap';

import { Brand, Home } from 'src/main/webapp/app/header/header-components';
import Header from 'src/main/webapp/app/header/header';

describe('Header', () => {
  let mountedWrapper;

  const userProps = {
    menuOpen: false,
    top10Click: () => {}
  };

  const wrapper = (props = userProps) => {
    if (!mountedWrapper) {
      mountedWrapper = shallow(<Header {...props} />);
    }
    return mountedWrapper;
  };

  beforeEach(() => {});

  // All tests will go here
  it('Renders a Header component in dev profile with LoadingBar, Navbar, Nav and dev ribbon.', () => {
    const component = wrapper();
    // the created snapshot must be committed to source control
    expect(component).toMatchSnapshot();
    const navbar = component.find(Navbar);
    expect(navbar.length).toEqual(1);
    expect(navbar.find(Brand).length).toEqual(1);
    const nav = component.find(Nav);
    expect(nav.length).toEqual(1);
    expect(nav.find(Home).length).toEqual(1);
    const ribbon = component.find('.ribbon.dev');
    expect(ribbon.length).toEqual(1);
  });

  it('Renders a Header component in prod profile with LoadingBar, Navbar, Nav.', () => {
    const component = wrapper(userProps);
    // the created snapshot must be committed to source control
    expect(component).toMatchSnapshot();
    const navbar = component.find(Navbar);
    expect(navbar.length).toEqual(1);
    expect(navbar.find(Brand).length).toEqual(1);
    const nav = component.find(Nav);
    expect(nav.length).toEqual(1);
    expect(nav.find(Home).length).toEqual(1);
    const ribbon = component.find('.ribbon.dev');
    expect(ribbon.length).toEqual(0);
  });

  it('Renders a Header component in prod profile with logged in User', () => {
    const nav = wrapper(userProps).find(Nav);
  });
});
