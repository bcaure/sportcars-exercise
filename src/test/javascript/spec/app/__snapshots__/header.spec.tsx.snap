// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Renders a Header component in dev profile with LoadingBar, Navbar, Nav and dev ribbon. 1`] = `
<div
  id="app-header"
>
  <Navbar
    class="jh-navbar"
    dark={true}
    expand="sm"
    fixed="top"
    tag="nav"
  >
    <NavbarToggler
      aria-label="Menu"
      onClick={[Function]}
      tag="button"
      type="button"
    />
    <Component />
    <Collapse
      appear={false}
      enter={true}
      exit={true}
      in={false}
      isOpen={false}
      mountOnEnter={false}
      navbar={true}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      tag="div"
      timeout={350}
      unmountOnExit={false}
    >
      <Nav
        class="ml-auto"
        id="header-tabs"
        navbar={true}
        tag="ul"
        vertical={false}
      >
        <Component
          onClick={[Function]}
        />
        <Component
          onClick={[Function]}
        />
        <Component
          onClick={[Function]}
        />
        <Component />
      </Nav>
    </Collapse>
  </Navbar>
</div>
`;

exports[`Header Renders a Header component in prod profile with LoadingBar, Navbar, Nav. 1`] = `
<div
  id="app-header"
>
  <Navbar
    class="jh-navbar"
    dark={true}
    expand="sm"
    fixed="top"
    tag="nav"
  >
    <NavbarToggler
      aria-label="Menu"
      onClick={[Function]}
      tag="button"
      type="button"
    />
    <Component />
    <Collapse
      appear={false}
      enter={true}
      exit={true}
      in={false}
      isOpen={false}
      mountOnEnter={false}
      navbar={true}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      tag="div"
      timeout={350}
      unmountOnExit={false}
    >
      <Nav
        class="ml-auto"
        id="header-tabs"
        navbar={true}
        tag="ul"
        vertical={false}
      >
        <Component
          onClick={[Function]}
        />
        <Component
          onClick={[Function]}
        />
        <Component
          onClick={[Function]}
        />
        <Component />
      </Nav>
    </Collapse>
  </Navbar>
</div>
`;
